### PR TITLE
Native toolset dependency blob storage format

### PR DESCRIPTION
**Pull request by chcosta**
Additional implementation doc for handling native toolset dependencies and how we'll store them.  

This is dependent on agreement for https://github.com/dotnet/arcade/pull/129.  Discussion there may drive this in a different direction, but I want to provide this additional doc as insight into how I'm reasoning about some of the implementation.

### CODE REVIEW

File: Documentation/Projects/NativeDependencies/NativeDependencyBlobStorageFormat.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+The "installer" is responsible for downloading the native asset and laying it out in the folder specified.  It will also download any shims required for using the native asset.
+
+**Why is each native dependency required to have an "installer", why isn't the local repo handling unzipping and laying out the assets?**
+
+I think that this model will allow us to be a bit more flexible in the types of dependencies that we install and provide a method for non-xcopy deployable dependencies to be installed in the future.
+
+**Is maintaining installers going to be a lot of maintenance?**
+
+I don't **think** so.  It should be a one-time cost for each asset and adding a new dependency should largely be a copy-paste type update with some version number updates.  If we decide to largely break away from how we layout assets or something else, we can use the "api version" to version these changes.
+
```
Comment (markwilkie) [inline Documentation/Projects/NativeDependencies/NativeDependencyBlobStorageFormat.md]: I wonder if we should provide a library to access the blob storage so we can change the formats later if/as needed?

Comment (chcosta) [reply Documentation/Projects/NativeDependencies/NativeDependencyBlobStorageFormat.md]: We could also redirect to a standard xcopy specific installation script here which is provided in the root of blob storage or directly from a git repo.  I'm not a fan of having to duplicate / maintain a bunch of nearly identical installers but I want the entry point semantics to be consistent.

Comment (chcosta) [reply Documentation/Projects/NativeDependencies/NativeDependencyBlobStorageFormat.md]: maybe even the entry point installers should be defined in a git repo vs living in blob storage.

Comment (chcosta) [reply Documentation/Projects/NativeDependencies/NativeDependencyBlobStorageFormat.md]: Repo seems like a better place for the installers / shims.  Blob storage would just contain the assets.

**Comment by chcosta** [comment]
I'm warming to the idea of defining the installers / shims in a git repo.  Moving that direction will make this doc largely irrelevant since the blob storage format will just be an Azure storage location with the assets.

**Pull request closed without merging**